src/EnergyPlus/General.hh

```diff
@@ -252,6 +252,175 @@ namespace General {
         XRes = XTemp;
     }
 
+    // A second version that does not require a payload -- use lambdas
+    template <typename Function>
+    void SolveRoot(EnergyPlusData &state,
```

Comment:
Aha! This doesn't need to be a `template` anymore, which is another win. The Function argument can be replaced with `std::function<Real64(Real64)> &f` (assuming that the function has one `Real64` argument UA and everything else is bound through the lambda, is that right).

Reply:
Oh yeah!  Great, yeah, that's right, it will always be simply `Real64 f(Real64 x)` with all the params going through the lambda capture.

Reply:
That is the beauty of lambda.

Reply:
OK, got this in place.  I had to declare it as a `const std::function` to get it to compile. Sandbox: https://godbolt.org/z/584Y74Tde

Reply:
Good to know. Did not know that.